cmd/juju/machine/upgradeseries.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machine
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/os/series"
+	"gopkg.in/juju/names.v2"
+
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// Actions
+const (
+	PrepareCommand  = "prepare"
+	CompleteCommand = "complete"
+)
+
+// NewUpgradeSeriesCommand returns a command which upgrades the series of
+// an application or machine.
+func NewUpgradeSeriesCommand() cmd.Command {
+	return modelcmd.Wrap(&upgradeSeriesCommand{})
+}
+
+type upgradeMachineSeriesAPI interface {
+}
+
+// upgradeSeriesCommand is responsible for updating the series of an application or machine.
+type upgradeSeriesCommand struct {
+	modelcmd.ModelCommandBase
+	// modelcmd.IAASOnlyCommand
+
+	// upgradeMachineSeriesClient upgradeMachineSeriesAPI
+
+	prepCommand   string
+	force         bool
+	machineNumber string
+	series        string
+}
+
+var upgradeSeriesDoc = `
+Upgrade a machine's operating system series.
+
+upgrade-series allows users to perform a managed upgrade of the operating system
+series of a machine. This command is performed in two steps; prepare and complete.
+
+The "prepare" step notifies Juju that a series upgrade is taking place for a given
+machine and as such Juju guards that machine against operations that would
+interfere with the upgrade process.
+
+The "complete" step notifies juju that the managed upgrade has been successfully completed.
+
+It should be noted that once the prepare command is issued there is no way to
+cancel or abort the process. Once you commit to prepare you must complete the
+process or you will end up with an unusable machine!
+
+The requested series must be explicitly supported by all charms deployed to
+the specified machine. To override this constraint the --force option may be used.
+
+The --force option should be used with caution since using a charm on a machine
+running an unsupported series may cause unexpected behavior. Alternately, if the
+requested series is supported in later revisions of the charm, upgrade-charm can
+run beforehand.
+
+Examples:
+	juju upgrade-series prepare <machine> <series>
+        juju upgrade-series prepare <machine> <series> --force
```

Comment:
nit: there's an extra space so juju doesn't line up in the examples.

Reply:
drive by on this as well